programs/mpl-core/src/plugins/immutable_metadata.rs

```diff
@@ -0,0 +1,33 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::program_error::ProgramError;
+
+use crate::state::DataBlob;
+
+use super::{PluginValidation, PluginValidationContext, ValidationResult};
+
+/// The immutable metadata plugin allows its authority to prevent plugin's meta from changing.
+/// The default authority for this plugin is None.
+
+#[repr(C)]
+#[derive(Clone, BorshSerialize, BorshDeserialize, Debug, PartialEq, Eq, Default)]
+pub struct ImmutableMetadata {}
+
+impl DataBlob for ImmutableMetadata {
+    fn get_initial_size() -> usize {
+        0
+    }
+
+    fn get_size(&self) -> usize {
+        0
+    }
+}
+
+impl PluginValidation for ImmutableMetadata {
+    /// Validate the update lifecycle action.
+    fn validate_update(
+        &self,
+        _ctx: &PluginValidationContext,
+    ) -> Result<ValidationResult, ProgramError> {
+        Ok(ValidationResult::Rejected)
```

Comment:
Hmm one alternative that would be nice for user is if this instead just returned an error like `MplError::ImmutableMetadata`.  wdyt? cc @blockiosaurus ?

Reply:
Updated the PR except for this part

Reply:
I'm hesitant to add errors now that we have ForceApprove. I think this would be handled by the bubble-up error ticket we have.